Jenkinsfile

```diff
@@ -1,91 +1,117 @@
+          name: 'KEYTABID'
+        ),
+        string(
+          defaultValue: 'v0.2',
+          description: 'Version of shared libraries to use',
+          name: 'LIBRARIES_VERSION'
+        ),
+        string(
+          defaultValue: 'https://github.com/RedHat-MultiArch-QE/multiarch-ci-libraries',
+          description: 'Repo for shared libraries',
+          name: 'LIBRARIES_REPO'
+        ),
+        string(
+          defaultValue: '',
+          description: 'Version of repo for tests to run.',
+          name: 'TEST_VERSION'
+        ),
+        string(
+          defaultValue: null
+          description: 'Repo for tests to run. If left blank, the current repo is assumed (*note* this default will only work for multibranch pipelines).',
+          name: 'TEST_REPO'
+        ),
+        string(
+          defaultValue: 'tests',
+          description: 'Directory containing tests to run. Should at least one of the follow: an ansible-playbooks directory containing one or more test directories each of which having a playbook.yml, a scripts directory containing one or more test directories each of which having a run-test.sh',
+          name: 'TEST_DIR'
+        )
+      ]
+    )
+  ]
 )
 
 List arches = params.ARCHES.tokenize(',')
 
 library changelog: false,
-        identifier: "multiarch-ci-libraries@${params.LIBRARIES_BRANCH}",
-        retriever: legacySCM([$class: 'GitSCM', branches: [[name: "*/${params.LIBRARIES_BRANCH}"]],
-                      doGenerateSubmoduleConfigurations: false,
-                      extensions: [], submoduleCfg: [],
-                      userRemoteConfigs:
-                              [[url: "${params.LIBRARIES_REPO}"]]])
+identifier: "multiarch-ci-libraries@${params.LIBRARIES_VERSION}",
+retriever: modernSCM([$class: 'GitSCMSource',remote: "${params.LIBRARIES_REPO}"])
 
 parallelMultiArchTest(
-        arches,
-        provisioningConfig.create(params),
+  arches,
+  provisioningConfig.create(),
   { Slave slave ->
-    /*******************************************************************/
-    /* TEST BODY                                                       */
-    /* @param slave               Name of the provisioned host.        */
-    /* @param arch                Architecture of the provisioned host */
-    /*******************************************************************/
-      dir('test') {
-          stage ('Download Test Files') {
-              checkout scm
-          }
+    /*********************************************************/
+    /* TEST BODY                                             */
+    /* @param slave               Provisioned slave details. */
+    /*********************************************************/
+    dir('test') {
+      stage ('Download Test Files') {
+        if (params.TEST_REPO) {
+          git url: params.TEST_REPO, branch: params.TEST_VERSION, changelog: false
```

Comment:
does this work if TEST_VERSION = '' as specified in defaults?

Reply:
good question - I will test this